src/Sentry.AspNetCore/ScopeExtensions.cs

```diff
@@ -68,29 +69,26 @@ public static void Populate(this Scope scope, HttpContext context, SentryAspNetC
             try
             {
                 var routeData = context.GetRouteData();
-                if (routeData != null)
+                var controller = routeData.Values["controller"]?.ToString();
+                var action = routeData.Values["action"]?.ToString();
+                var area = routeData.Values["area"]?.ToString();
+
+                if (controller != null)
+                {
+                    scope.SetTag("route.controller", controller);
+                }
+
+                if (action != null)
                 {
-                    var controller = routeData.Values["controller"]?.ToString();
-                    var action = routeData.Values["action"]?.ToString();
-                    var area = routeData.Values["area"]?.ToString();
-
-                    if (controller != null)
-                    {
-                        scope.SetTag("route.controller", controller);
-                    }
-                    if (action != null)
-                    {
-                        scope.SetTag("route.action", action);
-                    }
-                    if (area != null)
-                    {
-                        scope.SetTag("route.area", area);
-                    }
-
-                    scope.TransactionName = area == null
-                        ? $"{controller}.{action}"
-                        : $"{area}.{controller}.{action}";
+                    scope.SetTag("route.action", action);
                 }
+
+                if (area != null)
+                {
+                    scope.SetTag("route.area", area);
+                }
+
+                scope.TransactionName = context.TryGetRouteTemplate() ?? context.Request.Path;
```

Comment:
when would we fallback to Path here? Maybe better to leave it out is what I've heard.
If we can't get a proper template, we shoudn't fallback to the whole path

Reply:
We fallback to path if we can't get the route from endpoint routing or from MVC.
This makes it so that if you have a custom middleware that checks for request path (e.g. `if (context.Request.Path == "/health"`) then it at least shows up properly. I felt that it's more useful.